This is style cleanup of this one file

chick opened the pull request:
- Removed unused imports
- public vals that were collections have had types made explicit
- Some overridden functions now make their types explicit
- Some unused variable in matches are made underscores
- destTimescope in matches was shadowing the match source,renamed it
- Changed map whose return was not used to be foreach
- A couple of map functions were simplified by just passing function
  - instead of map (innerOverridingPokes(_))
  - becomes    map (innerOverridingPokes)
- A couple of spelling fixes
- used nonEmpty instead of .length > 0
 There are two remaining problems, both have ToDos

chick commented:
I got inspired while reading this code in IntelliJ, you can take it or leave it, but this makes things look a lot cleaner in the intellij editor

ducky64 commented on src/main/scala/chisel3/tester/ThreadedBackend.scala:
```diff
@@ -257,7 +260,7 @@ trait ThreadedBackend {
     // has not been overridden.
     // Conflicting pokes will be detected by poke checking.
-    signalPeeks.toSeq.map { case (signal, peeks) =>
+      signalPeeks.toSeq.foreach { case (signal, peeks) =>
```
Why the extra indent?

chick replied:
An accident, I'll fix this.

chick commented on src/main/scala/chisel3/tester/ThreadedBackend.scala:
```diff
@@ -32,7 +33,7 @@ trait ThreadedBackend {
   sealed trait HasOverridingPokes extends BaseTimescope {
     // List of timescopes overriding a signal, including ones closed / reverted on this timestep
-    val overridingPokes = mutable.HashMap[Data, mutable.ListBuffer[Timescope]]()
+    val overridingPokes: mutable.HashMap[Data, mutable.ListBuffer[Timescope]] = new mutable.HashMap
```
All public methods should have explicit type annotations. Type inference may break encapsulation in these cases, because it depends on internal method and class details. Without an explicit type, a change to the internals of a method or val could alter the public API of the class without warning, potentially breaking client code. Explicit type annotations can also help to improve compile times.  - From scala-lang

ducky64 replied:
I see, though these aren't meant to be public APIs, they're strictly internal data structures, though not marked as such yet. Feel free to make these internal classes protected or something, if that's possible.
My preference is the rule of DRY is more important than strict adherence to some set of rules (which are of questionable applicability here), absent some overriding concern.

chick replied:
I can make them private

chick commented on src/main/scala/chisel3/tester/ThreadedBackend.scala:
```diff
@@ -86,14 +87,16 @@ trait ThreadedBackend {
       val prefix: Seq[(BaseTimescope, Int)] = destTimescope match {
-        case destTimescope if destTimescope == startTimescope => Seq()
-        case destTimescope: HasParent => getLinearPath(startTimescope, destTimescope.parentTimescope, destTimescope.parentActionId)
+        case startDestTimescope if startDestTimescope == startTimescope => Seq()
```
I can't think of examples but I know Jack and Adam don't do it in firrtl which is heavily match oriented code. They are typically matching on expressions named e, their first order matches are ee and then eee and so on.

chick commented:
@ducky64 I dry'd out the mutable creation by making things private
I reverted the shadowed
I removed Steve Burns test case, which slipped in their accidentally
Please re-review
